python/paddle/nn/layer/common.py

```diff
@@ -348,6 +337,59 @@ def forward(self, input):
             data_format=self._data_format)
 
 
+class AlphaDropout(layers.Layer):
+    """
+    AlphaDropout Layer.
+    This interface is used to construct a callable object of the `AlphaDropout` class.
+    Alpha Dropout is a type of Dropout that maintains the self-normalizing property. For an input with
+    zero mean and unit standard deviation, the output of Alpha Dropout maintains the original mean and
+    standard deviation of the input. Alpha Dropout fits well to SELU activate function by randomly setting
+    activations to the negative saturation value.
+
+    For more information, please refer to:
+    [Self-Normalizing Neural Networks](https://arxiv.org/abs/1706.02515)
+    `paddle.nn.functional.alpha_dropout`
+
+    In dygraph mode, please use ``eval()`` to switch to evaluation mode, where dropout is disabled.
+
+    Parameters:
+        p (float, optional): Probability of setting units to zero. Default: 0.5
+        name (str|None): A name for this layer(optional). If set None, the layer
+                         will be named automatically.
+
+    Shape:
+        - input: N-D tensor.
+        - output: N-D tensor, the same shape as input.
+    Examples:
+        .. code-block:: python
+            import paddle
+            import numpy as np
+
+            paddle.disable_static()
+            x = np.array([[-1, 1], [-1, 1]]).astype('float32')
+            x = paddle.to_tensor(x)
+            m = paddle.nn.AlphaDropout(p=0.5)
+            y_train = m(x)
+            m.eval()  # switch the model to test phase
+            y_test = m(x)
+            print(x.numpy())
+            print(y_train.numpy())
+            # [[-0.10721093, 1.6655989 ], [-0.7791938, -0.7791938]] (randomly)
+            print(y_test.numpy())
+   """
+
+    def __init__(self, p=0.5, name=None):
+        super(AlphaDropout, self).__init__()
+        self.p = p
+        self.training = _dygraph_tracer()._train_mode
```

Comment:
`Layer` now has `train()`/`eval()`, this can be removed

Reply:
Done